app/models/event.rb

```diff
@@ -128,11 +141,11 @@ def icon_key
   end
 
   def editable?
-    ['note', 'activity'].include?(kind)
+    ['note', 'activity'].include?(kind) || !((self.chatwoot_message? || self.wpp_connect_message? ) && self.done?)
```

Comment:
Hmmm acho que tem algumas formas mais simples de fazer esses if.
Segue algumas sugestões:


1
```suggestion

  def editable?
    return true if ['note', 'activity'].include?(kind)
    return true if ['chatwoot_message', 'wpp_connect_message'].include?(kind) && !done?
    return false
  end

```


2
```suggestion

  def always_editable?
    ['note', 'activity'].include?(kind)
  end

  def editable?
    always_editable? || !done?
  end

```

Reply:
Achei a primeira opção mais fácil de entender, irei commitar ele mesmo. Valeu pelas sugestões.